src/sentry/workflow_engine/endpoints/validators/base/workflow.py

```diff
@@ -1,22 +1,103 @@
+from typing import Any
+
+from django.db import router, transaction
 from rest_framework import serializers
 
+from sentry import audit_log
 from sentry.api.serializers.rest_framework import CamelSnakeSerializer
-from sentry.workflow_engine.endpoints.validators.base import BaseDataConditionGroupValidator
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.endpoints.validators.base import (
+    BaseActionValidator,
+    BaseDataConditionGroupValidator,
+)
 from sentry.workflow_engine.endpoints.validators.utils import validate_json_schema
-from sentry.workflow_engine.models import Workflow
+from sentry.workflow_engine.models import (
+    DataConditionGroupAction,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+
+DataConditionGroupData = dict[str, Any]
+ActionData = dict[str, Any]
 
 
 class WorkflowValidator(CamelSnakeSerializer):
     name = serializers.CharField(required=True, max_length=256)
     enabled = serializers.BooleanField(required=False, default=True)
     config = serializers.JSONField(required=False)
     triggers = BaseDataConditionGroupValidator(required=False)
-    action_filters = serializers.ListField(child=BaseDataConditionGroupValidator(), required=False)
+    action_filters = serializers.ListField(required=False)
 
     # TODO - Need to improve the following fields (validate them in db)
     organization_id = serializers.IntegerField(required=True)
     environment_id = serializers.IntegerField(required=False)
 
+    def _split_action_and_condition_group(
+        self, action_filter: dict[str, Any]
+    ) -> tuple[ActionData, DataConditionGroupData]:
+        action = action_filter.get("action", None)
+
+        if not action:
+            raise serializers.ValidationError("An action is required in the actionFilter")
+
+        condition_group_keys = action_filter.keys() - action.keys()
+        condition_group = {k: action_filter[k] for k in condition_group_keys}
+
+        return action, condition_group
+
     def validate_config(self, value):
         schema = Workflow.config_schema
         return validate_json_schema(value, schema)
+
+    def validate_action_filters(self, value):
+        for action_filter in value:
+            action, condition_group = self._split_action_and_condition_group(action_filter)
+            BaseDataConditionGroupValidator(data=condition_group).is_valid(raise_exception=True)
+            BaseActionValidator(data=action).is_valid(raise_exception=True)
+
+        return value
+
+    def create(self, validated_value: dict[str, Any]) -> Workflow:
+        condition_group_validator = BaseDataConditionGroupValidator()
```

Comment:
should the Workflow validator automatically convert `DataConditionGroup.logic_type` of `ANY` into `ANY_SHORT_CIRCUIT`? there's some discussion here about how to handle it in the FE https://github.com/getsentry/sentry/pull/89233#discussion_r2047257627

Reply:
i don't feel like it should auto-convert that. It probably could based on the `type` but it would add some artificial constraints that i'm not sure i agree with.

we should _not_ convert `any` to `any_short_circuit` though as they are specifically used in different places for different reasons.